python_modules/libraries/dagster-k8s/dagster_k8s/job.py

```diff
@@ -352,37 +354,51 @@ def config_type_run_launcher(cls):
                     description="Whether the launched Kubernetes Jobs and Pods should fail if the Dagster run fails",
                 ),
             },
-            DagsterK8sJobConfig.config_type_job(),
         )
 
     @classmethod
     def config_type_job(cls):
-        """Configuration intended to be set at pipeline execution time."""
+        """Configuration intended to be set when creating a k8s job (e.g. in an executor that runs
+        each op in its own k8s job, or a run launcher that create a k8s job for the run worker).
+        Shares most of the schema with the container_context, but for back-compat reasons,
+        'namespace' is called 'job_namespace'."""
+
+        container_context_config_type = DagsterK8sJobConfig.config_type_container_context().copy()
+
+        # Some launchers and executors add this as `job_namespace` instead
+        del container_context_config_type["namespace"]
+
+        return merge_dicts(
+            {
+                "job_image": Field(
+                    Noneable(StringSource),
+                    is_required=False,
+                    description="Docker image to use for launched Jobs. If this field is empty, "
+                    "the image that was used to originally load the Dagster repository will be used."
+                    '(Ex: "mycompany.com/dagster-k8s-image:latest").',
+                ),
+            },
+            container_context_config_type,
+        )
```

Comment:
confused here - could the `namespace` delete happen where `job_namespace` is added? deleting here and adding elsewhere feels brittle. Or is not part of the config schema?